src/components/GeneOntologyNetwork.js

```diff
@@ -4,23 +4,58 @@ import coseBilkent from 'cytoscape-cose-bilkent';
 
 cytoscape.use(coseBilkent);
 
+function createTooltip(position, content) {
+	const div = document.createElement('div');
+	const rootElem = document.getElementsByClassName('rootContainer')[0];
+	if (rootElem.offsetWidth - position.x > 300) position.x += 50;
+	else position.x -= 350;
+	div.style.background = 'gray';
+	div.style.position = 'absolute';
+	div.style.color = 'white';
+	div.innerHTML = content;
+	div.style.top = position.y + 'px';
+	div.style.fontFamily = 'arial';
+	div.style.left = position.x + 'px';
+	div.style.padding = '10px';
+	div.style.width = '310px';
+	div.style.border = '1px solid black';
+	div.style.borderRadius = '8px';
+	document.body.append(div);
+	return div;
+}
+
 function GeneOntologyNetwork({ data }) {
 	useEffect(() => {
 		const elements = [];
 		data.forEach(el => {
```

Comment:
Separate out the `cytoscape` compatible node array generation logic to a separate function.